pkg/kvstore/etcd.go

```diff
@@ -584,7 +592,18 @@ func connectEtcdClient(ctx context.Context, config *client.Config, cfgPath strin
 	ec.controllers.UpdateController("kvstore-etcd-session-renew",
 		controller.ControllerParams{
 			DoFunc: func(ctx context.Context) error {
-				return ec.renewSession()
+				// Ignore the controller context here. Use the
+				// etcd client context instead, as this is
+				// renewing the session for that client.
+				// In the event of controller stop we'd expect
+				// the client context to be cancelled anyway
+				// and using the etcd context should generally
+				// integrate better with any code on the etcd
+				// client side.
+				parentCtx := ec.client.Ctx()
+				timedCtx, cancel := context.WithTimeout(parentCtx, statusCheckTimeout)
```

Comment:
We could potentially consider using something like `ClusterSizeDependantInterval()` to scale this based on the cluster size.